DOCS-2578 Track Android Errors Edits (#96)
DataDog/dd-sdk-android-gradle-plugin: docs/upload_mapping_file.md

## Overview

Error Tracking processes errors collected from the RUM Android SDK. 

Enable Android Crash Reporting and Error Tracking to get comprehensive crash reports and error trends with Real User Monitoring. With this feature, you can access:

- Aggregated Android crash dashboards and attributes
- Deobfuscated Android crash reports
- Trend analysis with Android error tracking

Your crash reports appear in [**Error Tracking**][1].

## Setup

If you have not set up the Android SDK yet, follow the [in-app setup instructions][2] or see the [Android RUM setup documentation][3].

> 1. Download the latest version of the [RUM Android SDK][4].

Reviewer: Users don't download the Android SDK but reference it in their Gradle project. 
```suggestion
1. Add the latest version of the [RUM Android SDK][4] to your Gradle Dependencies.
```

> 2. Configure your application's `version`, `env`, and `service` when [initializing the SDK][5].

Reviewer: The `version` and `service` are captured automatically, so customers don't have to set it manually, but they need to set their `variant`.
```suggestion
2. Configure your application's `env`, and `variant` when [initializing the SDK][5].
```

> 3. Upload your Proguard/R8 mapping file to Datadog to access deobfuscated stack traces. 

Reviewer: ```suggestion
3. Run the gradle task to upload your Proguard/R8 mapping file to Datadog to access deobfuscated stack traces. 
```

For any given error, you can access to the file path, line number, as well as a code snippet for each frame of the related stack trace.

## Upload your mapping file

{{< tabs >}}
{{% tab "US" %}}

1. Add the [Android Gradle Plugin][1] to your Gradle project using the following code snippet.

   ```groovy
   // In your app's build.gradle script
   plugins {
       id("com.datadoghq.dd-sdk-android-gradle-plugin") version "x.y.z"
   }
   ```

2. [Create a dedicated Datadog API key][2] and export it as an environment variable named `DD_API_KEY` (alternatively, pass it as a task property).

> 3. Optionally, configure the plugin to upload files to the EU region by exporting two additional environment variables: `export DATADOG_SITE="datadoghq.eu"` and `export DATADOG_API_HOST="api.datadoghq.eu"`.

Reviewer: I must have missed when those lines were added, but we don't support those two environment variables in the Gradle plugin (that must be a copy-paste from the `datadog-ci` tool). 

```suggestion
3. Optionally, configure the plugin to upload files to the EU region by configuring the plugin in your Gradle script: 
    datadog {
        site = "EU1"
    }
```

> 4. Run the upload task after your obfuscated APK builds:

Reviewer: This sentence feels weird, the APK is being built, it's not building itself :/

Author: ```suggestion
4. Run the upload task after your obfuscated APK starts being built:
```

    
   ```bash
   ./gradlew uploadMappingRelease
   ```

**Note**: If your project uses additional flavors, the plugin provides an upload task for each variant with obfuscation enabled. In this case, initialize the RUM Android SDK with a proper variant name (the necessary API is available in versions `1.8.0` and later).

[1]: https://github.com/DataDog/dd-sdk-android-gradle-plugin
[2]: https://app.datadoghq.com/account/settings#api

{{% /tab %}}
{{% tab "EU" %}}
1. Add the [Android Gradle Plugin][1] to your Gradle project using the following code snippet.

   ```groovy
   // In your app's build.gradle script
   plugins {
       id("com.datadoghq.dd-sdk-android-gradle-plugin") version "x.y.z"
   }
   ```

2. [Create a dedicated Datadog API key][2] and export it as an environment variable named `DD_API_KEY` (alternatively, pass it as a task property).
3. Configure the plugin to use the EU region by adding the following snippet in your app's `build.gradle` script file:

   ```groovy
   datadog {

>        site = "EU"

Reviewer: ```suggestion
       site = "EU1"
```

## Overview

Error Tracking processes errors collected from the RUM Android SDK. 

Enable Android Crash Reporting and Error Tracking to get comprehensive crash reports and error trends with Real User Monitoring. With this feature, you can access:

- Aggregated Android crash dashboards and attributes
- Deobfuscated Android crash reports
- Trend analysis with Android error tracking

Your crash reports appear in [**Error Tracking**][1].

## Setup

If you have not set up the Android SDK yet, follow the [in-app setup instructions][2] or see the [Android RUM setup documentation][3].

1. Add the latest version of the [RUM Android SDK][4] to your Gradle dependencies.
2. Configure your application's `env` and `variant` when [initializing the SDK][5].
3. Run the Gradle task to upload your Proguard/R8 mapping file to Datadog in order to access deobfuscated stack traces. 

> For any given error, you can access to the file path, line number, as well as a code snippet for each frame of the related stack trace.

Reviewer 2: ```suggestion
For any given error, you can access the file path, line number, and a code snippet for each frame of the related stack trace.
```

## Upload your mapping file

{{< tabs >}}
{{% tab "US" %}}

1. Add the [Android Gradle Plugin][1] to your Gradle project using the following code snippet.

   ```groovy
   // In your app's build.gradle script
   plugins {
       id("com.datadoghq.dd-sdk-android-gradle-plugin") version "x.y.z"
   }
   ```

2. [Create a dedicated Datadog API key][2] and export it as an environment variable named `DD_API_KEY` (alternatively, pass it as a task property).

> 3. Optionally, configure the plugin to upload files to the EU region by configuring the plugin in your Gradle script:

Reviewer 2: ```suggestion
3. Optionally, configure the plugin to upload files to the EU region by configuring the plugin in your `build.gradle` script:
```

   
   ```
   datadog {
       site = "EU1"
   }
   ```

> 4. Run the upload task after your obfuscated APK starts being built:

Reviewer 2: ```suggestion
4. Run the upload task after your obfuscated APK builds:
```

}
```

The task configuration for this variant is merged from all three flavor configurations provided in the following order:

1. `bar`
2. `foo`
3. `fooBar`

This resolves the final value for the `versionName` property as `fooBar`.

| Property name              | Description                                                                                                                                                                                               |
|----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `versionName`              | The version name of the application.                                                                                                                                                                      |

Reviewer: ```suggestion
| `versionName`              | The version name of the application (by default, the version declared in the `android` block of your `build.gradle` script).                                                                                                               |
```

> | `serviceName`              | The service name of the application.                                                                                                                                                                      |

Reviewer: ```suggestion
| `serviceName`              | The service name of the application (by default, the package name of your application as declared in the `android` block of your `build.gradle` script).                                                                                                                          |
```

> | `site`                     | The Datadog site to upload your data to (US, US3, US5, EU, or GOV)..                                                                                                                                       |

Reviewer: ```suggestion
| `site`                     | The Datadog site to upload your data to (US1, US3, US5, EU1, or US1_FED).                                                                                                                                       |
```
